File: src/main/webapp/scripts/game.js
function Game() {
  this.idGame;
  this.tries = [];
  this.target;
}

Game.prototype.on_result_received = function(idTry, sequence, result) {
  this.tries.push({SequenceNumber: idTry, InputNumbers: sequence, Result: result});

  this.notify_target();
}

Game.prototype.on_reset_game = function() {
	  this.tries = [];
	  
	  this.notify_target();
	}

Game.prototype.add_target = function(o) {
	this.target = o;
}


Game.prototype.notify_target = function() {
   this.target.notify(this);
}

Game.prototype.setIdGame = function(o){
	this.idGame = o;
}

Game.prototype.getIdGame = function(o){
	return this.idGame;
}

Game.prototype.getUsername = function(o){
	return this.username;
}

File: src/main/java/it/xpug/mastermind/main/Game.java
package it.xpug.mastermind.main;

public class Game {
	private String username;
	private String sequence;
	private String status;

	public Game(String username,String sequence, String status){
		this.username = username;
		this.sequence = sequence;
		this.status = status;
	}
	
	public String username(){
		return this.username;
	}
	
	public String status(){
		return this.status;
	}
	
	public String sequence(){
		return this.sequence;
	}
}
